### PR TITLE
Refactored tests

### PR DESCRIPTION
**Pull request by matejak**
* Made tests to use `pytest`, which makes them safer (esp. in regard with monkey patching) and more readable.
* Reduced duplication in the test code.
* Fixed some bugs.

### CODE REVIEW

**Comment by jan-cerny** [comment]
Please rebase it against RHEL 7.

**Comment by matejak** [comment]
Rebased agains rhel7-branch.

File: tests/test_common.py [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+
+    kwargs = {
+        "stdout": mock_subprocess.PIPE,
+        "stderr": mock_subprocess.PIPE,
+    }
+
+    return expected_args, kwargs
+
+
+def test_run_oscap_remediate_profile_only_test(mock_subprocess, monkeypatch):
```
[the remark this answers is not in the document]
Comment (matejak) [reply tests/test_common.py]: Fixed by rebase.

File: tests/test_rule_handling.py [hunk 2]
```diff
@@ -0,0 +1,560 @@
...
+
+    # / mount point missing --> one error
+    assert len(messages) == 1
+    assert messages[0].type == common.MESSAGE_TYPE_FATAL
+
+    # error has to mention the mount point
+    assert "/" in messages[0].text
+
+
+def get_messages_for_defaults_options(
```
[the remark this answers is not in the document]
Comment (matejak) [reply tests/test_rule_handling.py]: I have split this function and changed naming a little bit, please check it out again.

File: tests/test_rule_handling.py [hunk 3]
```diff
@@ -0,0 +1,560 @@
...
+    for message in messages:
+        if message.type == common.MESSAGE_TYPE_INFO:
+            assert "/" in message.text
+            assert "'noauto'" in message.text
+        elif message.type == common.MESSAGE_TYPE_FATAL:
+            assert "/tmp" in message.text
+            assert "'nodev'" not in message.text
+
+
+def test_evaluation_passwd_minlen_no_passwd(rule_data, ksdata_mock, storage_mock):
```
[the remark this answers is not in the document]
Comment (matejak) [reply tests/test_rule_handling.py]: Good point, I have implemented this.

**Comment by dahaic** [comment]
Thank you for the fixes, Matej! Looks great now. ACK

**Comment by matejak** [comment]
Damn I think that I discovered it - you meant contents `content_handling_test.py`. I will re-add it in another PR.

**Pull request merged**